src/Microsoft.AspNetCore.Razor.Evolution/RazorTemplateEngineResult.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Razor.Evolution
+{
+    public struct RazorTemplateEngineResult
```

Comment:
We need both the `RazorCodeDocument` and `RazorCSharpDocument` to produce a failed Razor compilation result. It doesn't feel intuitive for `GenerateCode` to return a code document (from which the consumer to get at the csharp document). I'd rather we leave this type in.